lightbluetent/society.py

```diff
@@ -26,6 +28,109 @@ def welcome(uid):
     if society.logo == current_app.config["DEFAULT_LOGO"]:
         has_logo = False
 
+    meeting = AttendeeMeeting(society.bbb_id, society.attendee_pw)
+    running = meeting.is_running()
+
+    if request.method == "POST":
+
+        # We should only be able to submit a POST request if the meeting is running.
+        # TODO: there may be circumstances where the meeting was running when this page was
+        # loaded, but it is no longer running when we submit this POST request. How should
+        # we handle this, if at all? Currently we abort(500).
+
+        full_name = request.form.get("full_name", "").strip()
+
+        errors = {}
+        if len(full_name) <= 1:
+            errors["full_name"] = "That name is too short."
+
+        if errors:
+            return render_template("society/welcome.html", page_title=f"{ society.name }",
+                           society=society, desc_paragraphs=desc_paragraphs,
+                           sessions_data=sessions_data, has_logo=has_logo, running=running,
+                           errors=errors)
+
+        if not running:
+            abort(500)
+
+        if full_name != "":
+            url = meeting.get_join_url(full_name)
+
+            # TODO: should we be logging this?
+            current_app.logger.info(f"Attendee '{ full_name }' joined stall for '{ society.name }', bbb_id: '{ society.bbb_id }'")
+
+            return redirect(url)
+
     return render_template("society/welcome.html", page_title=f"{ society.name }",
                            society=society, desc_paragraphs=desc_paragraphs,
-                           sessions_data=sessions_data, has_logo=has_logo)
+                           sessions_data=sessions_data, has_logo=has_logo, running=running,
+                           errors={})
+
+
+# Check if a meeting is running. If it's not, create it. Redirect to the URL to
+# join that meeting as a moderator with the provided name.
+@bp.route("/<uid>/begin", methods=("GET", "POST"))
+@auth_decorator
+def begin_session(uid):
+
+    society = Society.query.filter_by(uid=uid).first()
+
+    if not society:
+        return abort(404)
+
+    crsid = auth_decorator.principal
+    user = User.query.filter_by(crsid=crsid).first()
+    if society not in user.societies:
+        abort(403)
+
+    meeting = ModeratorMeeting(society.name,
```

Comment:
There is one change I request here, we should add another parameter to the API call that is `moderatorOnlyMessage` and set that to something like "To invite others into this session, share your stall link: https://freshers.srcf.net/s/balhsblash". Since it does not depend on the society/user params, we can hard code this in.